pi-top/pi-topOS-Web-Portal
# Fix duplicate SSIDs showing up in onboarding Wi-Fi menu

Ticket: https://pi-top.atlassian.net/browse/OS-1191

#### Summary

Currently, our app doesn't differentiate networks with  the same SSID but running on different frequencies (2.4GHz & 5GHz); same thing happens with networks that have multiple associated access points, our wifi manager & backend will just report all the networks that the wireless card can detect and the frontend app will display a network with the same name multiple times.

For example, the network `VM8387981` appears at least 3 times in this image:

![image](https://user-images.githubusercontent.com/14126805/150788534-6a6ddaaa-4940-41e1-b214-697e16a51720.png)


This PR introduces changes that:
- Differentiate 5G networks from regular 2.5G networks: backend adds a `[5G]` suffix to the ssid of the reported networks to the frontend app.
- In the case of a network being  reported multiple times because the backend detects multiple access points, the network with the higher signal level will be used in case the user wants to connect to it.
- Updates the frontend onboarding app to use the `bssid` instead of the `ssid` to identify a particular network.
- Re-enables (partially) python test runs on CI

#### Preview

![image](https://user-images.githubusercontent.com/14126805/150787867-848d0302-6738-4c09-b73d-452a8eada6bd.png)

Notice the networks `VTR-5845960` and `VTR-5845960 [5G]`. These would have appeared as duplicates in the previous implementation.

#### How to test?

1. Download latest 'deb' artifacts from the `Debian Package Build / build-debian-package` workflow.
2. Unzip the file and scp the `.deb` files into your pi-top
3. ssh into your pi-top and install the deb files with `sudo apt install ./*.deb` (running from the folder where you have the files)
4. Restart the web-portal service with `sudo systemctl restart pt-os-web-portal`
5. Load the onboarding and go to the wifi page either in your pi-top or via a browser connected to the same network as the device... e.g.: you could access `http://192.168.64.1/onboarding/wifi` if you're connected through the display-port cable.
6. Make sure that there are no duplicate SSIDs and that 5G networks have a `[5G]` suffix in their names. 
7. Try to connect to your network to make sure that works.
